dask/array/tests/test_slicing.py

```diff
@@ -421,6 +421,13 @@ def test_slicing_and_chunks():
     assert t.chunks == ((8, 8), (6, 6))
 
 
+def test_slicing_and_unknown_chunks():
+    a = da.ones((10, 5), chunks=5)
+    a._chunks = ((np.nan, np.nan), (5,))
```

Comment:
Rather than setting the chunks to `nan`, I think it would be better to write some user code that we *know* results in unknown chunks (like slicing with a boolean):

```python
a = da.random.uniform(size=10, chunks=2)
a = a[a > 0.5]
```

Reply:
This example doesn't seem to result in unknown chunks, I tried this and it has no problems. There are certain situations where we have nan in the chunks but we have ways of solving this. I'm not sure what's the right example that triggers the type of unknown chunks that causes this problem.